### PR TITLE
Fixes for issues #11 and #12

### PR DESCRIPTION
**Pull request by jperedadnr**
- Added an thread pool with 2 threads
- Added connection timeouts

### CODE REVIEW

File: src/main/java/com/gluonhq/impl/maps/ImageRetriever.java [hunk 1]
```diff
@@ -84,20 +86,24 @@
         }
     }
 
+    private final static Executor EXECUTOR = Executors.newFixedThreadPool(2, runnable -> {
+        Thread thread = new Thread(runnable);
+        thread.setDaemon(true);
```
Comment (johanvos) [inline src/main/java/com/gluonhq/impl/maps/ImageRetriever.java]: Is this what we want?

File: src/main/java/com/gluonhq/impl/maps/ImageRetriever.java [hunk 2]
```diff
@@ -174,28 +180,46 @@ public void cacheImage(String url, int zoom, long i, long j) {
...
+                openConnection = url.openConnection();
+                openConnection.setConnectTimeout(TIMEOUT);
+                openConnection.setReadTimeout(TIMEOUT);
+            } catch (IOException ex) {
+                logger.log(Level.SEVERE, null, ex);
+                return;
+            }
+            InputStream inputStream = null;
+            FileOutputStream fos = null;
+            try {
```
Comment (eugener) [inline src/main/java/com/gluonhq/impl/maps/ImageRetriever.java]: Using `try-with-resources` could eliminate a lot of code in `finally` block

Comment (jperedadnr) [reply src/main/java/com/gluonhq/impl/maps/ImageRetriever.java]: There is a reason for avoiding try-with-resources (GVM related).

Comment (eugener) [reply src/main/java/com/gluonhq/impl/maps/ImageRetriever.java]: That explains it! :)

**Pull request merged**